src/features/views/components/ViewDataTable/columnTypes/SurveyOptionsColumnType.tsx

```diff
@@ -37,6 +37,13 @@ export default class SurveyOptionsColumnType
       renderCell: (params: GridRenderCellParams) => {
         return <Cell cell={params.row[params.field]} />;
       },
+      sortComparator: (v1: string[][], v2: string[][]) => {
+        const v1n =
+          v1 == null || v1.length == 0 ? 0 : -(v1[v1.length - 1]?.length ?? 0);
+        const v2n =
+          v2 == null || v2.length == 0 ? 0 : -(v2[v2.length - 1]?.length ?? 0);
+        return v1n - v2n;
```

Comment:
After reading the code a few times (and a bit of googling) I understand that this `-(v1[v1.length - 1]?.length ?? 0)` can return a negative number, or -0 which evaluates to 0. Clever and cool way to write, and I got to learn about how +0 and -0 are treated like 0 in javascript in most cases! 

But that I was having a bit of difficulty reading this code, makes me think that if there was an issue here in the future it might be a bit of an unneccessary struggle to debug it. I think that it would benefit from being written a bit "less clever" with a few more lines, similar to my suggestion of creating a small util function like in the previous comment. 

Would you wanna do this?

Reply:
Hi, sure I'll change it, it's a bit of a bad habit to sneak in long ternaries whenever I can.

The idea with `v1[v1.length - 1]?.length` is that `v1.length` might be `-1`, so using `?.` instead of `.` will return `null` instead of throwing a error. `x ?? y` returns `x` if it's not `null` or `undefined`, or else return `y`.

Reply:
Turns out it was completely unnecessary here and in `SurveyResponseColumnType.tsx` since the length property is already checked before so I removed them.